maven-embedder/src/main/java/org/apache/maven/cli/configuration/SettingsXmlConfigurationProcessor.java

```diff
@@ -90,15 +90,15 @@ public void process( CliRequest cliRequest )
         throws Exception
     {
         CommandLine commandLine = cliRequest.getCommandLine();
-        String workingDirectory = cliRequest.getWorkingDirectory();
+        String multiModuleDirectory = cliRequest.getMultiModuleProjectDirectory().toString();
         MavenExecutionRequest request = cliRequest.getRequest();
 
         File userSettingsFile;
 
         if ( commandLine.hasOption( CLIManager.ALTERNATE_USER_SETTINGS ) )
         {
             userSettingsFile = new File( commandLine.getOptionValue( CLIManager.ALTERNATE_USER_SETTINGS ) );
-            userSettingsFile = resolveFile( userSettingsFile, workingDirectory );
+            userSettingsFile = resolveFile( userSettingsFile, multiModuleDirectory );
```

Comment:
Is it the expected behaviour to always resolve the path to settings file against the multi module root, even if it is specified directly through the command line, and not in .mvn/maven.config?

Reply:
No, it isn't, good catch! I suspected this solution would be too easy to be true, but hadn't realized this yet. It will have to move to the maven.config parser.